rfcs/012-add-session-ssh-support/012-add-session-ssh-support.md

```diff
@@ -0,0 +1,438 @@
+or perhaps the Renku CLI and this is entered via ssh. The ssh server then
+validates the generated token. More specifically, the token can be generated
+via keycloak and in the validation process, the ssh server requests keycloak to
+confirm the token provided is valid.
+
+Some open source software exists which supports ssh login via the Authorization
+Code Flow; interestingly it was developed in the context of HPC compute resource
+access. It is a modest component which could conceivably be supported if Renku
+opts to support this; however, it does not appear to have a large backing and it
+is written primarily in C.
+
+It is also worth noting that use of these mechanisms is not recommended within
+VSCode; VSCode indicates that standard key based login mechanisms are preferred.
+
+## Comparison of different approaches
+
+The following comparison criteria have been chosen to compare the different
+approaches:
+
+- VS Code support: whether the solution has good integration with VS Code ssh
+  mechanisms
+- Efficient IP address usage: whether use of IP addresses scales up with the
+  number of user sessions
+- Keys linked to user rather than project: whether each user has a dedicated
+  key/set of keys which can be arbitrarily mapped to user sessions or whether
+  keys are bound to projects rather than users
+- Flexible monitoring support: whether we can easily create grab user session
+  data to perform analysis of user behaviour
+- Key management responsibilities with users: whether users need to take
+  responsibility for managing keys
+- New Renku components to be developed/supported: whether the solution involves
+  development/support of new entities in the Renku platform
+
+A table showing how the different approaches compare is shown below:
+
+|                                                | Ssh port exposed by session directly | Proxy jumphost    | MITM Proxy         | OAuth2/OIDC       |
+|------------------------------------------------|--------------------------------------|-------------------|--------------------|-------------------|
+| VS Code support                                | :heavy_check_mark:                   | :heavy_check_mark: | :heavy_check_mark: | :x: |
+| Efficient IP address usage                     | :x:                                  | :heavy_check_mark: | :heavy_check_mark: | :heavy_check_mark: |
+| Keys linked to user rather than project        | :x:                                  | :x:                | :heavy_check_mark:  | N/A |
+| Flexible monitoring support                    | :x:                                  | :x:                | :heavy_check_mark:  | :heavy_check_mark:                |
+| Key management responsibilities with users     | :heavy_check_mark:                   | :heavy_check_mark: | :heavy_check_mark:  | :x:                |
+| New Renku components to be developed/supported | :x:                                  | :x:                | :heavy_check_mark:  | :heavy_check_mark: |
```

Comment:
I would consider needing to develop new components a negative trait, so use the red X? :)